ui/app/adapters/ssh/ca-config.js

```diff
@@ -8,13 +8,37 @@ import { encodePath } from 'vault/utils/path-encoding-helpers';
 
 export default class SshCaConfig extends ApplicationAdapter {
   namespace = 'v1';
-  // For now this is only being used on the vault.cluster.secrets.backend.configuration route. This is a read-only route.
-  // Eventually, this will be used to create the ca config for the SSH secret backend, replacing the requests located on the secret-engine adapter.
+
   queryRecord(store, type, query) {
     const { backend } = query;
     return this.ajax(`${this.buildURL()}/${encodePath(backend)}/config/ca`, 'GET').then((resp) => {
       resp.id = backend;
+      resp.backend = backend;
       return resp;
     });
   }
+
+  createOrUpdate(store, type, snapshot) {
+    const serializer = store.serializerFor(type.modelName);
+    const data = serializer.serialize(snapshot);
+    const backend = snapshot.record.backend;
+    data.id = backend;
+    return this.ajax(`${this.buildURL()}/${backend}/config/ca`, 'POST', { data }).then(() => {
+      // ember data requires an id on the response
+      return data;
+    });
```

Comment:
What does the backend return here after posting to the `config/ca` endpoint? It feels a little weird to rely on stale data from pre-saving. Is there is additional information the backend returns from this response? Wondering if we should do something like this instead:
```suggestion
    return this.ajax(`${this.buildURL()}/${backend}/config/ca`, 'POST', { data }).then((resp) => {
     return {
     ...resp,
     id: backend
    });
```

Reply:
Good idea!